src/main/java/mshantadze/pages/ProductListingPage.java

```diff
@@ -9,22 +9,74 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.LinkedList;
 import java.util.List;
 
 public class ProductListingPage extends BasePage {
     private final static Logger LOGGER = LoggerFactory.getLogger(ProductListingPage.class);
-    WebDriver driver;
 
     public ProductListingPage(WebDriver driver) {
         super(driver);
-        this.driver = driver;
         PageFactory.initElements(driver, this);
     }
 
     @FindBy(xpath = "//div[@id = 'srp-river-results']//span[@role = 'heading']")
-    List<WebElement> productNameLbls;
+    private List<WebElement> productNameLbls;
 
-    public List<WebElement> getProductNames() {
-        return productNameLbls;
+    @FindBy(className = "srp-save-null-search__heading")
+    private WebElement noProductFoundLbl;
+
+    @FindBy(xpath = "//div[@id='srp-river-results']//span[@class='s-item__price']")
+    private List<WebElement> priceLbls;
+
+    public boolean doesProductNameContain(String productName) {
```

Comment:
Do you expect scenario when boolean will be false? If not, change this logic to validateSearchingResults with Assert